### ISSUE TITLE
When -p / --pid option is used, the executable should not have to be passed on the command line

### ISSUE THREAD

**Issue by VincentGibert** [SimonKagstrom/kcov#107]
Hi,

I'm using a build of kcov from a git snapshot at revision 157d013. The man page states the executable argument is optional if kcov is launched with `--pid`. However, if the executable isn't specified, kcov terminates with an error.

1. For instance, launch the snmpd daemon

        $ /usr/sbin/snmpd

2. When trying to attach it without giving an executable name, an error occurs

        $ ps -ef | grep snmpd
        root      8307      1  0 16:03 ?        00:00:00 /usr/sbin/snmpd
        $ kcov --pid=8307 /tmp/kcov_results
        kcov: error: Can't find or open 

3. If the executable name is given, this error doesn't show up

        $ kcov --pid=8307 /tmp/kcov_results /usr/sbin/snmpd

Best regards,

**Comment by SimonKagstrom** [comment]
Yes, it was meant to be this way. However, I don't think I ever implemented it, and then forgot about that particular problem :-).

Should be at least fairly easy to implement via dereferencing the /proc/$PID/exe symlink. The only problem I have with that is that it deepens the Linux-dependence further - not that it's possible to build/run anywhere else right now anyway (Issue #21).

**Comment by VincentGibert** [comment]
I suggest you to update kcov man page to describe the real behaviour until you find a way to implement it in a portable manner (at first sight, it seems really difficult to do so without using a `#ifdef` per platform).

**Comment by SimonKagstrom** [comment]
Yes, I agree. ifdefs are ugly, so I'd like to avoid them where possible. In this case, I don't think it's possible to implement in a portable manner, at least not that I'm aware of.

Anyway, I've merged your documentation fix and changed the title of this bug report so that it describes the enhancement we've discussed for -p/--pid. I'll keep this bug open until it has been implemented, although I don't think it's very important since it mainly is a convenience issue.
